src/app/vault/ciphers.component.ts

```diff
@@ -81,6 +89,79 @@ export class CiphersComponent extends BaseCiphersComponent implements OnDestroy
     this.didScroll = this.pagedCiphers.length > this.pageSize;
   }
 
+  async loadOrganizationNames(): Promise<Map<string, string>> {
+    const organizations = await this.organizationService.getAll();
+    const organizationNames = new Map<string, string>();
+
+    organizations.forEach((organization) => {
+      organizationNames.set(organization.id, organization.name);
+    });
+
+    return organizationNames;
+  }
+
+  getOrganizationById(c: CipherView): string {
+    return c.organizationId != null
+      ? this.organizationNames.get(c.organizationId)
+      : this.i18nService.t("myVault");
+  }
+
+  sortCiphers(sortBy: string): void {
+    if (sortBy === this.sortBy) {
+      this.sortedDescending = !this.sortedDescending;
+    } else {
+      this.sortBy = sortBy;
+      this.sortedDescending = true;
+    }
+
+    if (sortBy === "lastEdited") {
+      if (this.sortedDescending) {
+        this.ciphers.sort((a: CipherView, b: CipherView) => {
+          return b.revisionDate.valueOf() - a.revisionDate.valueOf();
+        });
+      } else {
+        this.ciphers.sort((a: CipherView, b: CipherView) => {
+          return a.revisionDate.valueOf() - b.revisionDate.valueOf();
+        });
+      }
+      this.showData = "lastEdited";
+    } else if (sortBy === "dateCreated") {
+      // TO-DO
+    } else if (sortBy === "owner") {
+      this.ciphers.sort((a: CipherView, b: CipherView) => {
+        const sortingValue1 =
+          a.organizationId == null ? "Personal" : this.organizationNames.get(a.organizationId);
+        const sortingValue2 =
+          b.organizationId == null ? "Personal" : this.organizationNames.get(b.organizationId);
+
+        if (this.sortedDescending) {
+          return sortingValue1.localeCompare(sortingValue2);
+        } else {
+          return sortingValue2.localeCompare(sortingValue1);
+        }
+      });
+      this.showData = "owner";
+    } else if (sortBy === "name") {
```

Comment:
I found that `CipherService` already has a `getLocaleSortingFunction` which will sort ciphers by name (with some extra logic for login items). You'll need to use that here. e.g. `this.ciphers.sort(this.cipherService.getLocaleSortingFunction)`.

To order it descending, I think you could just do the above and then `this.ciphers.reverse()`. It's simple, and [this SO post](https://stackoverflow.com/a/52030227) suggests that it's not as inefficient as you might think. But please test it and make sure it works as expected.

Reply:
You might consider always sorting ascending, and then calling `reverse()` at the end if you need it descending. Would clean up this method a lot.